src/lsp-server/src/backend.rs

```diff
@@ -165,18 +191,69 @@ impl LanguageServer for Backend {
             .await;
     }
 
-    async fn execute_command(&self, _: ExecuteCommandParams) -> Result<Option<Value>> {
-        self.client
-            .log_message(MessageType::INFO, "command executed!")
-            .await;
-
-        match self.client.apply_edit(WorkspaceEdit::default()).await {
-            Ok(res) if res.applied => self.client.log_message(MessageType::INFO, "applied").await,
-            Ok(_) => self.client.log_message(MessageType::INFO, "rejected").await,
-            Err(err) => self.client.log_message(MessageType::ERROR, err).await,
+    /// Executes a single command and returns the response. Def: [workspace/executeCommand](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#workspace_executeCommand)
+    ///
+    /// Commands implemented:
+    ///
+    /// * parse: returns multiple valid statements from a single sql code.
+    async fn execute_command(&self, command_params: ExecuteCommandParams) -> Result<Option<Value>> {
```

Comment:
Is there a strong reason to avoid the [`?` operator](https://doc.rust-lang.org/rust-by-example/std/result/question_mark.html) here? This all seems correct but the nested `match` and `if let`s might be easier to read if we used `?` to exit early with `Err`s?

Reply:
What do you think about this:

```rust
let value = ...map_err(|x| build_error(std::borrow::Cow::Borrowed("Error message.")))?;

...

fn build_error(message: std::borrow::Cow<'static, str>) -> tower_lsp::jsonrpc::Error {
    Error {
        code: ErrorCode::InternalError,
        message,
        data: None,
    }
}
```

In this way I can reduce the use of `match` a lot, but I have to return the custom `Error` because, unfortunately, the trait expects its custom `tower_lsp::jsonrpc::Result` as a return value.